modules/dyn.js

```diff
@@ -0,0 +1,242 @@
+// dyn.js
+
+import {
+  pullComments, pullCommCounter, pushComment, pushLike, pullLikes,
+} from './socialApi.js';
+
+export default class DynGrid {
+  header = document.getElementById('Header');
+
+  footer = document.getElementById('Footer');
+
+  dynamicGrid = document.getElementById('dynamicGrid');
+
+  genCard = (data, i, unqId, imgSrc, foodName) => {
+    const id = i + 1;
+    const cardContainer = document.createElement('div');
+    const cardPic = document.createElement('div');
+    const cardPicImg = document.createElement('img');
+    cardPicImg.src = `${imgSrc}`;
+    const cardMeta = document.createElement('div');
+    const cardInfo = document.createElement('div');
+    const cardTitle = document.createElement('div');
+    cardTitle.innerHTML = `${foodName}`;
+    cardPic.classList.add('card-pic');
+    cardMeta.classList.add('card-meta');
+    cardInfo.classList.add('card-info');
+    cardTitle.classList.add('card-title');
+    const cardSocials = document.createElement('div');
+    cardSocials.classList.add('card-socials');
+    const likeBtn = document.createElement('img');
+    likeBtn.classList.add('like-btn');
+    likeBtn.src = '../src/icons/like.png';
+    likeBtn.alt = 'Like button';
+    const likeCounter = document.createElement('div');
+    likeCounter.classList.add('like-counter');
+    const showLikeQt = async (data, i, unqId) => {
+      likeCounter.innerHTML = `${await Promise.resolve(pullLikes(unqId))} likes`;
+    };
+    showLikeQt(data, i, unqId);
+    likeBtn.addEventListener('click', async () => {
+      await Promise.resolve(pushLike(unqId));
+      likeCounter.innerHTML = `${await Promise.resolve(pullLikes(unqId))} likes`;
+    });
+    const commentBtn = document.createElement('div');
+    commentBtn.id = `${id}`;
+    commentBtn.innerHTML = 'Comments';
+    const reserveBtn = document.createElement('div');
+    reserveBtn.innerHTML = 'Reservations';
+    commentBtn.classList.add('comment-btn');
+    reserveBtn.classList.add('reserve-btn');
+
+    cardContainer.className = 'card-container';
+    cardContainer.id = `${unqId}`;
+    this.dynamicGrid.className = 'dynamic-grid';
+
+    cardPic.appendChild(cardPicImg);
+    cardSocials.appendChild(likeBtn);
+    cardSocials.appendChild(likeCounter);
+    cardInfo.appendChild(cardTitle);
+    cardInfo.appendChild(cardSocials);
+    cardMeta.appendChild(cardInfo);
+    cardMeta.appendChild(commentBtn);
+    cardMeta.appendChild(reserveBtn);
+    cardContainer.appendChild(cardPic);
+    cardContainer.appendChild(cardMeta);
+    this.dynamicGrid.appendChild(cardContainer);
+
+    commentBtn.addEventListener('click', () => {
+      this.showPopup(data, i);
+    });
+  }
+
+  paintComments = (id, commentBoard, newC) => {
```

Comment:
- [ ] On the comment pop-up please also list some of the dish information and make sure it matches the design on the wireframe. You can list dish ingredients, recipes, etc. This will give an overview to users of each dish.

### see the wireframe image

![image](https://user-images.githubusercontent.com/83803414/163669627-23005df2-9aed-48de-a4c6-3db76e1d5210.png)

Reply:
Done, thank you!